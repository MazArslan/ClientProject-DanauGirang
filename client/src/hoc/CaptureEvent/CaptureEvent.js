import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import Spinner from "../../components/UI/Spinner/Spinner";
import CaptureEventForm from "../../containers/User/CaptureEventForm/CaptureEventForm";
import useHttp from "../../hooks/http";

/**
 * Higher order component to wrap the Capture Event Form component
 * Enables the capture Event Form to be accessed by the user statically
 */
const CaptureEvent = () => {
  const { projectId, captureEventId } = useParams();
  const { error, data, sendRequest } = useHttp();
  const [tags, setTags] = useState();

  useEffect(() => {
    sendRequest(`http://localhost:3000/api/project/${projectId}/tags`, "GET");
  }, [projectId, sendRequest]);

  useEffect(() => {
    if (data !== null && !error) {
      setTags(data.projectTags);
    }
  }, [data, error, setTags]);

  let captureEventForm = <Spinner />;
  if (tags) {
    captureEventForm = (
      <CaptureEventForm tags={tags} captureEventId={captureEventId} />
    );
  }

  return <div>{captureEventForm}</div>;
};

export default CaptureEvent;
