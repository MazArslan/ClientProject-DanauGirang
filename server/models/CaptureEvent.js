const mongoose = require("mongoose");
var Schema = mongoose.Schema;

const CaptureEventSchema = new Schema({
	projectId: {
		type: mongoose.SchemaTypes.ObjectId,
		required: true,

		index: true
	},
	notes: {
		type: String,
		required: true
	},
	coordinates: {
		type: String,
		required: true
	},
	isComplete: {
		type: Boolean,
		required: true
	},
	createdOnDateTime: {
		// Mongoose is a bit weird in that it's datetime type is Date
		// See: https://www.oreilly.com/library/view/mongoose-for-application/9781782168195/ch03s02.html
		type: Date,
		required: true
	},
	tags: [
		{
			type: String
		}
	]
});

// Retrieved 23/2/20 https://stackoverflow.com/a/23100950
// Mongoose creates a collection name which is the model name all lowercase and pluralised
// So in this case 'captureevents'
// The third argument in the .model() function allows this convention to be overwritten
const captureEvents = mongoose.model(
	"CaptureEvent",
	CaptureEventSchema,
	"captureEvents"
);

module.exports = captureEvents;
