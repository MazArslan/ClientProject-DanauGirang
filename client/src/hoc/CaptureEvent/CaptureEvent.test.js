import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import CaptureEvent from "./CaptureEvent";

Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    projectId: "5e63b45c1c9d4400001bb204",
    captureEventId: "5e63b45c1c9d4400001ba312"
  })
}));

describe("<CaptureEventForm/>", () => {
  describe("component rendering", () => {
    const wrapper = shallow(<CaptureEvent />);

    it("should render shallow correctly with props", () => {
      expect(wrapper.exists()).toBe(true);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should render a spinner", () => {
      const spinner = wrapper.find("spinner");
      expect(spinner).toBeTruthy();
    });
  });
});
