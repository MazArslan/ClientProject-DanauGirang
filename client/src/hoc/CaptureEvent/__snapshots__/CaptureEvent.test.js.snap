// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaptureEventForm/> component rendering should render shallow correctly with props 1`] = `
<div>
  <spinner />
</div>
`;
